client/sites-dashboard-v2/sites-dataviews/sites-site-sort.tsx

```diff
@@ -0,0 +1,102 @@
+// Copied from client/a8c-for-agencies/sections/sites/site-sort/index.tsx as we don't have SitesDashboardContext here.
+import { Icon } from '@wordpress/icons';
+import classNames from 'classnames';
+import {
+	defaultSortIcon,
+	ascendingSortIcon,
+	descendingSortIcon,
+} from 'calypso/jetpack-cloud/sections/agency-dashboard/icons';
+import type { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+
+import 'calypso/a8c-for-agencies/sections/sites/site-sort/style.scss';
+
+const SORT_DIRECTION_ASC = 'asc';
+const SORT_DIRECTION_DESC = 'desc';
+
+// Mapping the columns to the site data keys
+const SITE_COLUMN_KEY_MAP: { [ key: string ]: string } = {
+	site: 'site',
+};
+
+interface SiteSortProps {
+	columnKey: string;
+	isLargeScreen?: boolean;
+	children?: React.ReactNode;
+	isSortable?: boolean;
+	dataViewsState: DataViewsState;
+	setDataViewsState: ( callback: ( prevState: DataViewsState ) => DataViewsState ) => void;
+}
+
+export const SiteSort = ( {
+	columnKey,
+	isLargeScreen,
+	children,
+	isSortable,
+	dataViewsState,
+	setDataViewsState,
+}: SiteSortProps ) => {
+	const { field, direction } = dataViewsState.sort;
+
+	const isDefault = field !== SITE_COLUMN_KEY_MAP?.[ columnKey ] || ! field || ! direction;
+
+	const setSort = () => {
+		const updatedSort = { ...dataViewsState.sort };
+		if ( isDefault ) {
+			updatedSort.field = SITE_COLUMN_KEY_MAP?.[ columnKey ];
+			updatedSort.direction = SORT_DIRECTION_ASC;
+		} else if ( direction === SORT_DIRECTION_ASC ) {
+			updatedSort.direction = SORT_DIRECTION_DESC;
+		} else if ( direction === SORT_DIRECTION_DESC ) {
+			updatedSort.direction = SORT_DIRECTION_ASC;
+		}
+
+		setDataViewsState( ( sitesViewState ) => ( {
+			...sitesViewState,
+			sort: updatedSort,
+		} ) );
+	};
+
+	const getSortIcon = () => {
+		if ( isDefault ) {
+			return defaultSortIcon;
+		} else if ( direction === SORT_DIRECTION_ASC ) {
+			return ascendingSortIcon;
+		} else if ( direction === SORT_DIRECTION_DESC ) {
+			return descendingSortIcon;
+		}
+		return defaultSortIcon;
+	};
+
+	if ( ! isSortable ) {
+		return <span className="site-sort">{ children }</span>;
+	}
+
+	const handleOnKeyDown = ( event: React.KeyboardEvent< HTMLDivElement > ) => {
+		if ( event.key === 'Enter' || event.key === ' ' ) {
```

Comment:
Minor thing I noticed:

The behavior of the space `event.key === ' '` is a bit weird, it scrolls the page as well as updates the sort:

https://github.com/Automattic/wp-calypso/assets/6586048/8da675ea-c85a-408c-a022-d116370b3f74

Enter works perfectly.